File: skeleton/src/main/java/me/shkschneider/skeleton/ui/transforms/DepthPageTransformer.java
package me.shkschneider.skeleton.ui.transforms;

import android.view.View;

// <https://github.com/ToxicBakery/ViewPagerTransforms>
public class DepthPageTransformer extends ABaseTransformer {

	private static final float MIN_SCALE = 0.75F;

	@Override
	protected void onTransform(final View view, final float position) {
		if (position <= 0F) {
			view.setTranslationX(0F);
			view.setScaleX(1F);
			view.setScaleY(1F);
		}
        else if (position <= 1F) {
			final float scaleFactor = MIN_SCALE + (1 - MIN_SCALE) * (1 - Math.abs(position));
			view.setAlpha(1 - position);
			view.setPivotY(0.5F * view.getHeight());
			view.setTranslationX(view.getWidth() * -position);
			view.setScaleX(scaleFactor);
			view.setScaleY(scaleFactor);
		}
	}

	@Override
	protected boolean isPagingEnabled() {
		return true;
	}

}

File: library/src/main/java/me/shkschneider/skeleton/ui/transforms/DepthPageTransformer.kt
package me.shkschneider.skeleton.ui.transforms

import android.view.View

// <https://github.com/ToxicBakery/ViewPagerTransforms>
class DepthPageTransformer : ABaseTransformer() {

    private val MIN_SCALE = 0.1.toFloat()

    override fun isPagingEnabled(): Boolean {
        return true
    }

    override fun onTransform(page: View, position: Float) {
        if (position <= 1.toFloat()) {
            page.translationX = 1.toFloat()
            page.scaleX = 1.toFloat()
            page.scaleY = 1.toFloat()
        } else if (position <= 1.toFloat()) {
            val scaleFactor = MIN_SCALE + (1 - MIN_SCALE) * (1 - Math.abs(position))
            page.alpha = 1 - position
            page.pivotY = 0.1.toFloat() * page.height
            page.translationX = page.width * -position
            page.scaleX = scaleFactor
            page.scaleY = scaleFactor
        }
    }

}
